### PR TITLE
Reduce frequency of automatically archiving tabs

### PR DESCRIPTION
**Pull request by EdwardJTL**
as a part of our efforts to reduce the cost of archived tabs, and after checking with @Ria-yu, we are limiting "automatically archiving" tabs to only happen when restoring from start up tabs.

### How was this tested?
- [ ] I tested this manually
- [ ] I added automated tests
- [x] Existing automated tests are enough to cover my changes

### CODE REVIEW

**Comment by EdwardJTL** [comment]
read more about what we are doing and why here: https://paper.dropbox.com/doc/Archived-Tabs-Storage-Design-Doc--BsHHQRgG6KgJ8nOBaWBbQqdfAg-2OAo9pYYfwpRHmkDZNA2w

File: Client/Frontend/Browser/Tabs/TabManager.swift [hunk 1]
```diff
@@ -511,24 +511,6 @@ class TabManager: NSObject, TabEventHandler, WKNavigationDelegate {
             selectedTab.lastExecutedTime = Date.nowMilliseconds()
         }
 
-        // Determine if any of the active tabs should be converted to archived tabs.
```
[the remark this answers is not in the document]
Comment (EdwardJTL) [reply Client/Frontend/Browser/Tabs/TabManager.swift:514]: In `TabManagerStore`, `restoreTabs(savedTabs: [SavedTab], clearIncognitoTabs: Bool, tabManager: TabManager)` goes through the loaded tabs and add them to TabManager if the tab should be archived

File: Client/Frontend/Browser/Tabs/TabManager.swift [hunk 2]
```diff
@@ -157,14 +153,6 @@ class TabManager: NSObject, TabEventHandler, WKNavigationDelegate {
                         self?.selectedTabURLPublisher.send($0)
                     }
             }
-
```
Comment (evancrow) [inline Client/Frontend/Browser/Tabs/TabManager.swift:160]: Do we need to notify the user that their preferences won't take effect until the relaunch? Similar to `FeatureFlags`?

Comment (EdwardJTL) [reply Client/Frontend/Browser/Tabs/TabManager.swift:160]: good point, i will check with ria

File: ClientTests/CardTests.swift [hunk 3]
```diff
@@ -488,9 +488,47 @@ class CardTests: XCTestCase {
...
+
+    // MARK: - Archived Tab Tests
     /// Add a tab last used over a week ago, check if it shows up in the archives.
     /// Default time threshold to archive a tab is 7 days.
-    func testTabsInArchivedSection() {
+    func testTabsInArchivedSection() throws {
+        throw XCTSkip(
+            "This test requires writing an archived tabs to storage and loading the app. API not available."
+        )
+
```
[the remark this answers is not in the document]
Comment (EdwardJTL) [reply ClientTests/CardTests.swift:531]: yes i'm sure :)

File: Client/Frontend/Browser/Tabs/TabManager.swift (hunk 2 above)
Comment (EdwardJTL) [reply Client/Frontend/Browser/Tabs/TabManager.swift:160]: updated the string. @chung-lai tagging you so we remember to update the localized strings

File: ClientTests/CardTests.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (EdwardJTL) [reply ClientTests/CardTests.swift:531]: this test ensures that when `updateAllTabDataAndSendNotifications` is called while the app is actively being used, tabs are archived. This functionality is no longer supported. instead, tabs are only archived when the start up tabs are loaded from disk. We need a new test that checks this start up behaviour. These tests are planned as more functionalities are added as a part of the feature change. I find that there is little value in writing up the new utils or coercing the existing utils at this point in the project given how much the tab storage infrastructure is going to change.

**Review by evancrow — APPROVED** [review]

**Pull request merged**